src/utils/build.js

```diff
@@ -321,171 +315,158 @@ const build = (zipPath, sourceZipPath, wdir) => {
+  wdir = wdir || process.cwd();
+  zipPath = zipPath || constants.BUILD_PATH;
+  sourceZipPath = sourceZipPath || constants.SOURCE_PATH;
+  const osTmpDir = await fse.realpath(os.tmpdir());
+  const tmpDir = path.join(
+    osTmpDir,
+    'zapier-' + crypto.randomBytes(4).toString('hex')
+  );
 
-      endSpinner();
-      startSpinner('Validating project');
-
-      return Promise.all([
-        _appCommandZapierWrapper(tmpDir, { command: 'validate' }),
-        Promise.resolve(rawDefinition)
-      ]);
-    })
-    .then(([validateResponse, rawDefinition]) => {
-      const errors = validateResponse.results;
-      if (errors.length) {
-        return Promise.resolve({
-          errors: {
-            validation: errors
-          }
-        });
-      }
+  maybeNotifyAboutOutdated();
 
-      // No need to mention specifically we're validating style checks as that's
-      //   implied from `zapier validate`, though it happens as a separate process
+  // make sure our directories are there
+  await ensureDir(tmpDir);
+  await ensureDir(constants.BUILD_DIR);
 
-      return callAPI('/style-check', {
-        skipDeployKey: true,
-        method: 'POST',
-        body: rawDefinition
-      });
-    })
-    .then(styleChecksResponse => {
-      const errors = styleChecksResponse.errors;
-      if (!_.isEmpty(errors)) {
-        if (global.argOpts.debug) {
-          console.log('\nErrors:');
-          console.log(errors);
-          console.log('');
-        }
+  startSpinner('Copying project to temp directory');
+  await copyDir(wdir, tmpDir, {
+    filter: process.env.SKIP_NPM_INSTALL
+      ? dir => !dir.includes('.zip')
+      : undefined
+  });
 
-        throw new Error(
-          'We hit some validation errors, try running `zapier validate` to see them!'
-        );
-      }
+  let output = {};
+  if (!process.env.SKIP_NPM_INSTALL) {
+    endSpinner();
+    startSpinner('Installing project dependencies');
+    output = await runCommand('npm', ['install', '--production'], {
+      cwd: tmpDir
+    });
+  }
+
+  // `npm install` may fail silently without returning a non-zero exit code, need to check further here
+  const corePath = path.join(
+    tmpDir,
+    'node_modules',
+    constants.PLATFORM_PACKAGE
+  );
+  if (!await fs.exists(corePath)) {
+    throw new Error(
+      'Could not install dependencies properly. Error log:\n' + output.stderr
+    );
+  }
+  endSpinner();
+
+  startSpinner('Applying entry point file');
+  // TODO: should this routine for include exist elsewhere?
+  const zapierWrapperBuf = await readFile(
+    path.join(
+      tmpDir,
+      'node_modules',
+      constants.PLATFORM_PACKAGE,
+      'include',
+      'zapierwrapper.js'
+    )
+  );
+  await writeFile(`${tmpDir}/zapierwrapper.js`, zapierWrapperBuf.toString());
+  endSpinner();
+
+  startSpinner('Building app definition.json');
+  const rawDefinition = (await _appCommandZapierWrapper(tmpDir, {
+    command: 'definition'
+  })).results;
```

Comment:
I think we need to remove the `.results` part, i.e.,
```js
  const rawDefinition = await _appCommandZapierWrapper(tmpDir, {
    command: 'definition'
  });
```
Otherwise, it will error when you do `rawDefinition.results` again.